stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -0,0 +1,195 @@
+        "stock.location.route",
+        string="Delivery Route",
+        ondelete="set null",
+        readonly=True,
+    )
+    # TODO rules should be a related field through 'delivery_route_id' and
+    # they should be limited to the flow config (current domain)
+    rule_ids = fields.One2many(
+        related="delivery_route_id.rule_ids",
+        string="Rules",
+    )
+    impacted_route_ids = fields.One2many(
+        comodel_name="stock.location.route",
+        compute="_compute_impacted_route_ids",
+        string="Impacted Routes",
+    )
+
+    def _default_warehouse_id(self):
+        warehouse = self.env["stock.warehouse"].search([])
+        if len(warehouse) == 1:
+            return warehouse
+
+    @api.depends("from_picking_type_id.default_location_src_id")
+    def _compute_from_location_src_id(self):
+        for record in self:
+            location = record.from_picking_type_id.default_location_src_id
+            if not location:
+                __, location = self.env["stock.warehouse"]._get_partner_locations()
+            record.from_location_src_id = location
+
+    @api.depends("from_picking_type_id.default_location_dest_id")
+    def _compute_from_location_dest_id(self):
+        for record in self:
+            location = record.from_picking_type_id.default_location_dest_id
+            if not location:
+                location, __ = self.env["stock.warehouse"]._get_partner_locations()
+            record.from_location_dest_id = location
+
+    def _compute_impacted_route_ids(self):
+        for flow in self:
+            rules = self.env["stock.rule"].search(
+                [
+                    ("picking_type_id", "=", flow.from_picking_type_id.id),
+                    ("route_id.flow_id", "=", False),
+                ]
+            )
+            flow.impacted_route_ids = rules.route_id
+
+    @api.onchange("warehouse_id")
+    def onchange_warehouse_id(self):
+        self.from_picking_type_id = self.warehouse_id.out_type_id
+
+    def _generate_delivery_route(self):
+        self.ensure_one()
+        if self.delivery_route_id:
+            return False
+        try:
+            with self.env.cr.savepoint():
+                # TODO delivery steps has to be set by the user (new wizard?)
+                self.warehouse_id.delivery_steps = self.delivery_steps
+                vals = self.warehouse_id.delivery_route_id.copy_data()[0]
+                raise Exception()
+        except Exception:
+            pass
+        vals.update(
+            {
+                "name": f"{self.warehouse_id.name}: {self.name}",
+                "active": True,
+                "flow_id": self.id,
+                # Give the priority to routes/rules not tied to a flow
+                # when the initial move is created by the procurement
+                "sequence": 1000,
+            }
+        )
+        for rule_vals in vals["rule_ids"]:
+            # Replace the initial outgoing picking type by the new one
```

Comment:
The complete flow needs dedicated picking types for each step

Reply:
Is it something we want every time or it's something we need for our project? Why can't we re-use an existing picking type for a pick or pack operation if the work to do is the same than another route or flow?

Reply:
Mostly every time. Especially that the number of steps can be different from the warehouse config. So the location src/dest of the warehouse default picking types won't fit anyway.

Reply:
In the rare case you would not want a new picking type, you can select an existing one before the route is generated

Reply:
got it